### PR TITLE
Solving warnings

### PR DESCRIPTION
**Pull request by dougolima**
# Description

In order to improve the code quality of the project, all warnings were solved.

Actions done:

- Documentation for all public classes
- Decreased classes visibility from public to internal (where it was possible)
- Stop using obsolete properties
- Turned off code validations for unit test and integration test projects

## How Has This Been Tested?

Unit tests and integration tests were executed

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests to cover my changes

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: src/KafkaFlow.Abstractions/Acks.cs [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+    /// <summary>Acks enum values</summary>
     public enum Acks
     {
+        /// <summary>Leader</summary>
         Leader,
+        
+        /// <summary>All</summary>
         All,
+        
+        /// <summary>None</summary>
```
Comment (filipeesch) [inline src/KafkaFlow.Abstractions/Acks.cs:12]: In WithAcks method there is a description of each one, you can remove from there and put it here

File: src/KafkaFlow.Abstractions/AutoOffsetReset.cs [hunk 2]
```diff
@@ -1,8 +1,12 @@
 namespace KafkaFlow
 {
+    /// <summary>AutoOffsetReset enum values</summary>
     public enum AutoOffsetReset
     {
+        /// <summary>Latest</summary>
         Latest,
+        
+        /// <summary>Earliest</summary>
```
Comment (filipeesch) [inline src/KafkaFlow.Abstractions/AutoOffsetReset.cs:9]: In WithAutoOffsetReset method there is a description of each one, you can remove from there and put it here

File: src/KafkaFlow.Abstractions/Configuration/IConsumerConfigurationBuilder.cs [hunk 3]
```diff
@@ -104,7 +110,7 @@ IConsumerConfigurationBuilder WithWorkDistributionStrategy<T>()
         IConsumerConfigurationBuilder WithAutoStoreOffsets();
 
         /// <summary>
-        /// The Handler or Middleware should call the <see cref="IMessageContext.StoreOffset()"/>
+        /// The handler or middleware should call the <see cref="IMessageContextConsumer.StoreOffset()"/>
```
Comment (filipeesch) [inline src/KafkaFlow.Abstractions/Configuration/IConsumerConfigurationBuilder.cs:113]: The client should call...

File: src/KafkaFlow.Abstractions/Acks.cs (hunk 1 above)
Comment (dougolima) [reply src/KafkaFlow.Abstractions/Acks.cs]: DONE

**Pull request merged**